### DISCUSSION TITLE
Copy a GitHub Project Board *and* the cards

### DISCUSSION THREAD

**Discussion by johncokos** [community/community#21981] [Projects and Issues]
<p>We’ve got a real need (as an educational institution) to create repositories for our students that, in addition to code and other materials, have a project board with a series of cards and tasks.</p>
<p>Ideally we want to have the students be able to clone or fork these repositories and have a fully setup project board with tasks already there for them.</p>
<p>I can see where we can copy a board (settings and automation) but not the cards.  Is there a simple way to do an export/import or import from a JSON file or something similar?</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/johncokos">@johncokos</a>! :wave:</p>
<p>There’s no way built into the web UI to copy across all the cards from a Project Board, sorry! We’re always working to improve GitHub, and we consider every suggestion we receive, so perhaps you’d like to submit a feature request <a href="https://support.github.com/contact/feedback">through our official product feedback form</a> so that our product team can see exactly how you’d like GitHub to work?</p>
<p>Our API allows for the fetching and creation of cards, so you could probably come up with something that would do the trick:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <a href="https://developer.github.com/v3/projects/cards/" target="_blank">GitHub Developer</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/19803303/181068025-d4b3093d-5363-4aef-9488-0d219b51b138.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_10x10.png">

<h3><a href="https://developer.github.com/v3/projects/cards/" target="_blank">Project cards</a></h3>

<p>Get started with one of our guides, or jump straight into the API documentation.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I’d be interested to hear if other members of the community had already come up with scripts to do this!</p>

**Comment by zer0Kerbal** [comment]
<p>as would I - really would love the ability to git the projects to local (with cards) and be able to copy them through either the file system or github app or the online repo.</p>

**Comment by DomVinyard** [comment]
<ul>
<li>one on this, would be really useful</li>
</ul>

**Reply by SallyMcGrath** [reply]
Fancy seeing you here! 😂

**Comment by lpmi-13** [comment]
<p>We had the same need and ended up working around it, using a github action workflow plus metadata file, which isnt as nice as just “copying”, but got us to the outcome we wanted.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/learning-as-code/learning-space-template">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/learning-as-code/learning-space-template" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/5070516/181068027-6bcef3c9-cc02-4164-9e3b-8f74af87da23.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/4/14bab0ba062ba0386a477f2ed863df98fb3a6085_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/4/14bab0ba062ba0386a477f2ed863df98fb3a6085_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/4/14bab0ba062ba0386a477f2ed863df98fb3a6085.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/4/14bab0ba062ba0386a477f2ed863df98fb3a6085_2_10x10.png"></div>

<h3><a href="https://github.com/learning-as-code/learning-space-template" target="_blank" rel="noopener nofollow ugc">GitHub - learning-as-code/learning-space-template: an example template to...</a></h3>

  <p>an example template to create a learning space for a single learner - GitHub - learning-as-code/learning-space-template: an example template to create a learning space for a single learner</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Reply by SallyMcGrath** [reply]
omg the gang's all here!
